CONTRIBUTING.md

```diff
@@ -16,3 +16,141 @@ You might be wondering what a contribution is. Here are some things that you can
 - Review [open Pull Requests](https://github.com/swapagarwal/swag-for-dev/pulls?q=is%3Apr+is%3Aopen+sort%3Aupdated-desc).
 
 Thanks again, we hope to see you soon! ❤
+
+## Let's start hacking!
+
+Clone this repository by running
+```sh
+git clone https://github.com/swapagarwal/swag-for-dev.git
+```
+
+### Start using NPM
+
+1. Ensure you are running an up-to-date version of [Node.js](https://nodejs.org/en/download/package-manager/) on your machine, and that you have npm installed.
+1. Open a terminal in the `swag-for-dev` directory. Type
+    ```sh
+    npm install
+    ```
+    into the terminal to install the dependencies. To start the webserver, type
+    ```sh
+    npm start
+    ```
+1. The website should open in a browser after it compiles, or you can view it by going to http://localhost:8000
+
+---
+
+1. To build a production release, type
+    ```sh
+    npm run build
+    ```
+1. Dist is now ready in the `swag-for-dev/dist` directory. You can use any web-server to preview it in your browser. One way is using `http-server` module directly from npm as follows
+    ```sh
+    npx http-server -p 8000 ./dist
+    ```
+
+### Start using Docker
+
+1. Ensure you are running an up-to-date version of [Docker](https://docs.docker.com/install/) on your machine.
+1. Open a terminal in the `swag-for-dev` directory. Type
+    ```sh
+    docker build -t devswag:dev --target base .
+    ```
+    to build a docker image `devswag:dev`.
+1. You now have to run this image with
+    ```sh
+    docker run -it --rm \
+    -v $(pwd)/data.json:/devswag/data.json \
+    -v $(pwd)/src:/devswag/src \
+    -v $(pwd)/gulpfile.js:/devswag/gulpfile.js \
+    -v $(pwd)/get-data.js:/devswag/get-data.js \
+    -v $(pwd)/dist:/devswag/dist \
+    -u $(id -u):$(id -g) \
+    -p 8000:8000 -p 35729:35729 devswag:dev
```

Comment:
It gets stuck at the watch task:
```
[05:09:17] Finished 'swag-img:optimize' after 4.13 s
[05:09:17] Finished 'swag-img' after 29 s
[05:09:17] Starting 'cachebust'...
[05:09:17] Finished 'cachebust' after 847 μs
[05:09:17] Starting 'pug'...
[05:09:18] Finished 'pug' after 305 ms
[05:09:18] Finished 'build' after 29 s
[05:09:18] Starting 'webserver'...
[05:09:18] Webserver started at http://0.0.0.0:8080
[05:09:18] Finished 'webserver' after 30 ms
[05:09:18] Starting 'watch'...
```

Reply:
It looks legit to me as it's watching for changes in the folder. Can you access http://127.0.0.1:8000?

Reply:
Unable to connect

Reply:
Fixed the documentation, default port was 8080.

Reply:
Not working for me.
@vikaspotluri123 Can you check once?

Reply:
Yep

Reply:
I ran into a bunch of permissions errors, but I found the bug... suggesting changes

Reply:
@swapagarwal can you try again? The port should be 8080